### PR TITLE
docs: add apisix integrates with hydra blog

### PR DESCRIPTION
**Pull request by guitu168**
add apisix integrates with hydra blog

Screenshots of the change:

### CODE REVIEW

File: website/blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+  - name: "Fei Han"
+    title: "Technical Writer"
+    url: "https://github.com/hf400159"
+    image_url: "https://github.com/hf400159.png"  
+keywords: 
+- Apache APISIX
+- API Gateway
+- Authentication
+- Hydra
+- Openid connect
```
Comment (SylviaBABY) [inline website/blog/2022/07/04/apisix-integrates-with-hydra.md]:
```suggestion
- OpenID Connect
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+[Ory Hydra](https://github.com/ory/hydra) 是支持 OAuth 2.0 和 OpenID Connect 协议的身份提供商之一，基于 OAuth 2.0 授权框架以及 Open ID Connect Core 1.0 框架实现，兼具开源与云原生特性。它可以与任何登录系统集成，通过 OAuth 2.0 Acces、Refresh 和 ID Tokens，使得第三方能够轻松访问你的 API，实现用户随时随地与任何应用程序的交互。
+
+Ory Hydra 采用 Go 语言开发，为几乎所有开发语言都提供了 SDK，包括 Dart、.NET、Go、Java、PHP、Python、Ruby、Rust 和 Typescript。它适用于任何登录系统，并且可以轻松自定义登录体验。
+
+## 简介
+
+OpenID 是一种集中认证模式，它是一个去中心化的身份认证系统。使用 OpenID 的好处是：用户只需要在一个 OpenID 身份提供方的网站上注册和登录，使用一份账户密码信息即可访问不同应用。
+
+通过 APISIX 支持的 `openid-connect` 插件，我们可以与支持 OpenID Connect 协议的认证程序集成。比如：[Ory Hydra]。更多信息请参考：[集中身份认证](https://apisix.apache.org/zh/blog/2021/08/25/using-the-apache-apisix-openid-connect-plugin-for-centralized-authentication/#%E4%BB%80%E4%B9%88%E6%98%AF%E8%BA%AB%E4%BB%BD%E8%AE%A4%E8%AF%81)。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/07/04/apisix-integrates-with-hydra.md]:
```suggestion
通过 APISIX 支持的 `openid-connect` 插件，我们可以与支持 OpenID Connect 协议的认证程序集成，比如 Ory Hydra。更多信息请参考：[集中身份认证](https://apisix.apache.org/zh/blog/2021/08/25/using-the-apache-apisix-openid-connect-plugin-for-centralized-authentication)。
```

File: website/blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+One of the biggest advantages of Ory Hydra is that it implements the OAuth and OpenID Connect standards instead of forcing you to use "Hydra user management" (login, logout, profile management, registration), a specific template engine, or a predefined front end.
+
+It allows to use the authentication mechanisms required by your program (token-based 2FA, SMS 2FA, etc.) and implement user management and login in your technology stack. Of course, you can also use existing solutions, such as [authboss](https://github.com/go-authboss/authboss). It gives you all the great features of OAuth 2.0 and OpenID Connect while being minimally intrusive to your business logic and technology stack.
+
+OAuth 2.0 can be used in many environments for various purposes. This list might help you decide if OAuth 2.0 and Hydra are the right fit for a use case:
```
Comment (SylviaBABY) [inline website/blog/2022/07/04/apisix-integrates-with-hydra.md]:
```suggestion
OAuth 2.0 can be used in many environments for various purposes. The following information may help you decide whether OAuth 2.0 and Hydra are suitable for a certain scenario:
```

File: website/blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+1. enable third-party solutions to access your APIs.
+2. be an Identity Provider like Google, Facebook, or Microsoft: OpenID Connect, and thus Hydra is a perfect fit.
```
Comment (SylviaBABY) [inline website/blog/2022/07/04/apisix-integrates-with-hydra.md]:
```suggestion
2. be an Identity Provider like Google, Facebook, or Microsoft.
```

File: website/blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+3. enable your browser, mobile, or wearable applications to access your APIs: Running an OAuth2 Provider can work great for this. You don't have to store passwords on the device and can revoke access tokens at any time.
+4. you want to limit what type of information your backend services can read from each other. For example, the comment service should only be allowed to fetch user profile updates but shouldn't be able to read user passwords.
+
+## Operation steps
+
+Next, I will show you how APISIX integrates with Hydra using a real example. In this example, Docker will be used to running the required environment. Please install [Docker](https://docs.docker.com/engine/install/) before doing this.
+
+### Step 1: Create and deploy the database
+
+For quick deployment of the test environment, we will use Docker to run PostgreSQL as Hydra's database. Using Docker to run the database in production is not recommended.
```
Comment (SylviaBABY) [inline website/blog/2022/07/04/apisix-integrates-with-hydra.md]:
```suggestion
For quick deployment of the test environment, we will use Docker to run PostgreSQL as Hydra's database. It's not recommended to use Docker to run the database in production.
```

File: website/blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+docker run \
+  --network hydraguide \
+  --name ory-hydra-example--postgres \
+  -e POSTGRES_USER=hydra \
+  -e POSTGRES_PASSWORD=secret \
+  -e POSTGRES_DB=hydra \
+  -d postgres:9.6
+```
+
+The above command will create a network named hydraguide and start a Postgres instance named ory-hydra-example--postgres which creates the database hydra, the user hydra, and the user password secret.
```
Comment (SylviaBABY) [inline website/blog/2022/07/04/apisix-integrates-with-hydra.md]:
```suggestion
The above command will create a network named `hydraguide` and start a Postgres instance named `ory-hydra-example--postgres` which creates the database `hydra`, the user `hydra`, and the user password `secret`.
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+  -d postgres:9.6
+```
+
+上述命令将创建一个名称为 `hydraguide` 的网络，并且启动一个名称为 `ory-hydra-example--postgres` 的 Postgres 实例，该实例创建了数据库 `hydra`，用户 `hydra` 和用户密码 `secret`。
+
+### 步骤二：部署 Hydra
+
+该步骤中将会把 `4444` 映射到 `5444` 和 `4445` 映射到 `5445` 端口，请确保这些端口未被占用。
+
+1. 系统密钥只能针对新数据库设置，不支持密钥轮换。 此密钥用于加密数据库，每次进程（重新）启动时都需要设置为相同的值。 你可以使用 /dev/urandom 生成密钥。但请确保在你定义它的时候，该密钥都必须相同。例如，你可以将值存储在某处:
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/07/04/apisix-integrates-with-hydra.md]:
```suggestion
1. 系统密钥只能针对新数据库设置，不支持密钥轮换。 此密钥用于加密数据库，每次进程（重新）启动时都需要设置为相同的值。 你可以使用 `/dev/urandom` 生成密钥。但请确保在你定义它的时候，该密钥都必须相同。例如，你可以将值存储在某处:
```

File: website/blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+docker run -d \
+  --name ory-hydra-example--consent \
+  -p 9020:3000 \
+  --network hydraguide \
+  -e HYDRA_ADMIN_URL=https://ory-hydra-example--hydra:4445 \
+  -e NODE_TLS_REJECT_UNAUTHORIZED=0 \
+  oryd/hydra-login-consent-node:v1.10.6
+```
+
+You can check if the program is working properly with the command:
```
Comment (SylviaBABY) [inline website/blog/2022/07/04/apisix-integrates-with-hydra.md]:
```suggestion
You can use the following command to check whether the program runs normally:
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+![网络错误/认证页面](https://static.apiseven.com/2022/blog/0704/1.png)
+
+2. 选择认证协议，并单击 `Allow Access`。
+
+![网络错误/选择页面](https://static.apiseven.com/2022/blog/0704/2.png)
+
+3. 验证成功之后，就可以访问到上游服务的页面。
+
+![网络错误/上游页面](https://static.apiseven.com/2022/blog/0704/3.png)
+
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/07/04/apisix-integrates-with-hydra.md]: Duplicate step 2, need to fix it. Use tab to indent pictures into each step should work perfectly for you.
image

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+  -e POSTGRES_PASSWORD=secret \
+  -e POSTGRES_DB=hydra \
+  -d postgres:9.6
+```
+
+上述命令将创建一个名称为 `hydraguide` 的网络，并且启动一个名称为 `ory-hydra-example--postgres` 的 Postgres 实例，该实例创建了数据库 `hydra`，用户 `hydra` 和用户密码 `secret`。
+
+### 步骤二：部署 Hydra
+
+该步骤中将会把 `4444` 映射到 `5444` 和 `4445` 映射到 `5445` 端口，请确保这些端口未被占用。
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/07/04/apisix-integrates-with-hydra.md]:
```suggestion
该步骤中会分别将 `4444` 和 `4445` 映射到 `5444` 和 `5445` 端口，请确保这些端口未被占用。
```

File: website/blog/2022/07/04/apisix-integrates-with-hydra.md [hunk 11]
```diff
@@ -0,0 +1,286 @@
...
+![network-error/Authentication page](https://static.apiseven.com/2022/blog/0704/1.png)
+
+2. Select the authentication protocol and click `Allow Access`.
+
+![network-error/select page](https://static.apiseven.com/2022/blog/0704/2.png)
+
+3. After successful verification, you can access the upstream service page.
+
+![network-error/upstream page](https://static.apiseven.com/2022/blog/0704/3.png)
+
```
Comment (yzeng25) [inline website/blog/2022/07/04/apisix-integrates-with-hydra.md]: ditto, see comments in zh-cn file.

**Review by yzeng25 — DISMISSED** [review]

**Review by SylviaBABY — APPROVED** [review]
LGTM

**Review by yzeng25 — APPROVED** [review]

**Pull request merged**